File: spoj/done/apaga.py
respostas = []
while(1):
	pilha = []
	resposta = ""
	substituidos = 0
	linha1 = raw_input().split()
	n = int(linha1[0])
	d = int(linha1[1])
	if n == 0 and d == 0:
		break
	c = raw_input()
	for numero in c:
		num = int(numero)
		while(len(pilha) != 0 and pilha[len(pilha) - 1] < num and substituidos < d):
			pilha.pop()
			substituidos += 1
		pilha.append(num)
	while(substituidos < d):
		pilha.pop()
		substituidos += 1
	for p in pilha:
		resposta += str(p)
	respostas.append(resposta)
for r in respostas:
	print r
	

File: spojbr/done/apaga.c
#include <stdio.h>
#include <stdlib.h>
#define MAX 100000

typedef struct{
    int topo;
    int valor[MAX];
}pilha;

void iniciapilha(pilha *p)
{
    p->topo=-1;
}


void push(pilha *p, float a)
{
    if(p->topo!=MAX-1)
    {
        p->topo++;
        p->valor[p->topo]=a;
    }
}

float pop(pilha *p)
{
    float ans;
    if(p->topo!=-1)
    {
        ans=p->valor[p->topo];
        p->topo--;
    }
    return ans;
}


int main(){
	
	pilha* p;
	int i, N, D, num, substituidos;
	char lixo, entrada[100000];
	
	p=malloc(sizeof(pilha));
	
while(1){
	iniciapilha(p);
	substituidos = 0;
	scanf("%d", &N);
	scanf("%d", &D);
	if(N == 0 && D == 0) return 0;
	scanf("%c", &lixo);
	for(i=0;i<N;i++){
		scanf("%c", &entrada[i]);
	}
	for(i=0;i<N;i++){
		num = (int)entrada[i] - 48;		
		while(p->topo != -1 && p->valor[p->topo] < num && substituidos < D){
			pop(p);
			substituidos++;
		}
		push(p, num);
	}
	while(substituidos < D){
		pop(p);
		substituidos++;
	}
	for(i=0;i<N-D;i++)  printf("%d", p->valor[i]);
	printf("\n");
}
	free(p);
	return 0;
}
